### ISSUE TITLE
IllegalArgumentException: Field 'xxx': type mismatch. Was DATE

### ISSUE THREAD

**Issue by beeender** [realm/realm-java#3188]
This kind of crash happened to @KynoYang a few times, but the crash logs are not exact the same:

Crash 1

```
java.lang.IllegalArgumentException: Field 'xxx': type mismatch. Was DATE, expected [STRING].
    at io.realm.RealmObjectSchema.getColumnIndices(RealmObjectSchema.java:618)
    at io.realm.RealmQuery.equalTo(RealmQuery.java:262)
    at io.realm.RealmQuery.equalTo(RealmQuery.java:249)
    at com.xxx.xx.data.xxx.getLatestMsg(xxx.java:80)
```

Crash 2

```
io.realm.exceptions.RealmError: Unrecoverable error. Data type mismatch in io_realm_internal_TableQuery.cpp line 886
    at io.realm.internal.TableQuery.nativeEqual(Native Method)
    at io.realm.internal.TableQuery.equalTo(TableQuery.java:350)
    at io.realm.RealmQuery.equalTo(RealmQuery.java:263)
    at io.realm.RealmQuery.equalTo(RealmQuery.java:249)
    at com.xx.xxx.data.xxxxxxx.getXXXByVid(xxx.java:98)
```

From the log we can see the first exception was thrown in the java, but the 2nd one was thrown from native code. This is abnormal. Since the both of them went through the same calling routine that the 2nd one did the java side type checking as well -- But it threw later.

It might indicate that we have a corrupted pointer that caused by a concurrency problem.

PS.: There are also a few other crashes from @KynoYang which indicate the similar problem. But those two are quite typical.

Might be fixed by https://github.com/realm/realm-core/pull/1958 . We need to monitor this after `v1.1.1` release.

**Comment by ironage** [comment]
Do we know what version of core this happened with?
Is it possible the user is misusing Realm across threads?
I will move this to waiting for user until we can get some more information.

**Comment by cmelchior** [comment]
Since we haven't seen any more reports of this, I will assume it was fixed by 1.1.1. Closing
